pkg/proxy/topology.go

```diff
@@ -160,19 +164,23 @@ func canUseTopology(endpoints []Endpoint, svcInfo ServicePort, nodeLabels map[st
 		}
 	}
 
-	zone, ok := nodeLabels[v1.LabelTopologyZone]
-	if !ok || zone == "" {
-		klog.V(2).InfoS("Skipping topology aware endpoint filtering since node is missing label", "label", v1.LabelTopologyZone)
-		return false
-	}
-
+	zone, foundZone := nodeLabels[v1.LabelTopologyZone]
 	hasEndpointForZone := false
 	for _, endpoint := range endpoints {
 		if !endpoint.IsReady() {
 			continue
 		}
+
+		// If any of the endpoints do not have zone hints, we bail out
 		if endpoint.ZoneHints().Len() == 0 {
-			klog.V(2).InfoS("Skipping topology aware endpoint filtering since one or more endpoints is missing a zone hint", "endpoint", endpoint)
+			return false
+		}
+
+		// If there is at least one zone hint, we check if there is a zone label, if there isn't one we bail out
```

Comment:
It took me awhile to understand what's happening, maybe 
```suggestion
		// If we've made it this far, we have endpoints with hints set. Now we check if there is a zone label, if there isn't one we log a warning and bail out
```

Reply:
applying